DifferentiationInterface/ext/DifferentiationInterfaceMooncakeExt/onearg.jl

```diff
@@ -0,0 +1,100 @@
+struct MooncakeOneArgPullbackPrep{Y,R} <: PullbackPrep
+    y_prototype::Y
+    rrule::R
+end
+
+function DI.prepare_pullback(
+    f, backend::AutoMooncake, x, ty::DI.Tangents, contexts::Vararg{Context,C}
+) where {C}
+    y = f(x, map(unwrap, contexts)...)
```

Comment:
Is it part of the interface of DI that `f` must not mutate either `x` or `map(unwrap, contexts)...)`?

Reply:
For now yes, because the only available context type is `Constant`. Later I'll add a second type called `Cache` with different properties.